File: Objective-C/TCMPPDemo/Utils/CustomBundle.m
//
//  CustomBundle.m
//  TCMPPDemo
//
//  Created by gavinjwxu on 2024/8/2.
//

#import "CustomBundle.h"
#import "LanguageManager.h"

@implementation CustomBundle

- (NSString *)localizedStringForKey:(NSString *)key value:(NSString *)value table:(NSString *)tableName {
    NSBundle *customBundle = [[LanguageManager shared] getBundle];
    return [customBundle localizedStringForKey:key value:value table:tableName];
}

@end

File: Swift/TCMPPDemo-Swift/Class/Utils/CustomBundle.swift
//
//  CustomBundle.swift
//  TCMPPDemo-Swift
//
//  Created by gavinjwxu on 2024/8/13.
//

import Foundation

class CustomBundle: Bundle {
    override func localizedString(forKey key: String, value: String?, table tableName: String?) -> String {
        let customBundle = LanguageManager.shared.getBundle()
        return customBundle?.localizedString(forKey: key, value: value, table: tableName) ?? Bundle.main.localizedString(forKey: key, value: value, table: tableName)
    }
}
